src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/DescendentsWalker.cs

```diff
@@ -114,18 +112,13 @@ private void IterateChildren(DependencyObject d)
             {
                 // Neither a FrameworkElement nor FrameworkContentElement.  See
                 //  if it's a Visual and if so walk the Visual collection
-                Visual v = d as Visual;
-                if (v != null)
+                if (d is Visual v)
                 {
                     WalkVisualChildren(v);
                 }
-                else
+                else if (d is Visual3D v3D)
                 {
-                    Visual3D v3D = d as Visual3D;
-                    if (v3D != null)
-                    {
-                        WalkVisualChildren(v3D);
-                    }
+                    WalkVisualChildren(v3D);
                 }
```

Comment:
Is this change fine ?

Reply:
It should be, as it doesn't change the original logic.